### PR TITLE
Add Dhcp6 Client/Server Support to Windows Driver

### PR DESCRIPTION
**Pull request by nibanks**

### CODE REVIEW

**Comment by jwhui** [comment]
@nibanks, I can see the value in this for the short-term.  Long-term, I think OpenThread should be able to integrate with external DHCPv6 services (i.e. Windows' implementations).

**Comment by nibanks** [comment]
@jwhui, I agree, but that won't happen until we integrate cross interface routing. When we bring that in, then we would switch out the DHCP.

File: examples/drivers/windows/otLwf/filter.h [hunk 1]
```diff
@@ -36,8 +36,11 @@
 #define _FILT_H
 
 // The maximum allowed addresses an OpenThread interface
-#define OT_MAX_ADDRESSES 10
-#define OT_MAX_AUTO_ADDRESSES (OT_MAX_ADDRESSES - 4)
+#if OPENTHREAD_ENABLE_DHCP6_CLIENT
```
Comment (nibanks) [inline examples/drivers/windows/otLwf/filter.h]: So you are saying that if `DHCP6_SERVER` is enabled, then I need to account for `OPENTHREAD_CONFIG_NUM_DHCP_PREFIXES` ALOCs?

Comment (librasungirl) [inline examples/drivers/windows/otLwf/filter.h]: Yes if`otCachedAddr[]` is used to cache all the Unicast Addresses the thread device could have

Comment (jwhui) [inline examples/drivers/windows/otLwf/filter.h]: @nibanks, are you planning on updating this PR?

Comment (nibanks) [inline examples/drivers/windows/otLwf/filter.h]: Yes, I will try to get it updated now.

**Comment by kwangeom** [comment]
could anybody let me know how to enable DHCPv6 client on the cli mtd side?
I want border router to assign dhcpv6 address using dnsmasq to cli mtd. To do this, i have set 
/* Define to 1 if you want to enable DHCPv6 Client */
#define OPENTHREAD_ENABLE_DHCP6_CLIENT 1
With this change, I expected cli mtd to send Solict message, but could not see it from wireshark capture.
Are there anything else I should do more to enable dhcpv6 client in cli mtd side?
My setup is : border router - ncp(leader and router) - cli mtd (end device).

Best Regards,

Kwang

**Comment by jwhui** [comment]
@kwangeom, you can enable DHCPv6 Client support by adding `DHCP6_CLIENT=1` to your build command line. For example:
```
$ make -f examples/Makefile-nrf52840 DHCP6_CLIENT=1
```

**Comment by kwangeom** [comment]
@jwhui  With this change, still I cannot see the Solicit message from the end device. Digging the issue, I found DHCPv6 is not allowed from the Border Router TLV. But  I cannot find where to set enable DHCPv6 support from border router and NCP side. Could you please guide me how to set this field?

**Comment by jwhui** [comment]
@kwangeom, if you are using wpanctl, you can use the `wpanctl add-prefix` command:
```
$ wpanctl add-prefix --help
Syntax:
   add-prefix [args] <prefix>
Options:
   -h/--help                 Print Help
   -t/--timeout              Set timeout period [ms]
   -P/--priority             Indicate prefix priority (default is 0 or medium) [(>0 for high, 0 for medium, <0 for low)]
   -l/--length               Set the prefix length (default is 64) [in bits]
   -s/--stable               Indicate the prefix is part of stable Network Data (default is off)
   -f/--preferred            Set the prefix flag "preferred"
   -a/--slaac                Set the prefix flag "SLAAC"
   -d/--dhcp                 Set the prefix flag "dhcp"
   -c/--configure            Set the prefix flag "configure"
   -r/--default-route        Set the prefix flag "default-route"
   -o/--on-mesh              Set the prefix flag "on-mesh"
```

For example, to add a new on-mesh prefix with dhcp support, you can do the following:
```
$ wpanctl add-prefix --stable --preferred --dhcp --default-route --on-mesh <ipv6-prefix>
```

BTW, this PR is closed. If you would like to discuss further, please open a new GitHub issue so we can track it. Thanks.

**Comment by kwangeom** [comment]
@jwhui  Thanks so much.

**Pull request merged**